src/microscope_laser_autofocus/Objective.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using WDI;
+
+using Zaber.Motion;
+using Zaber.Motion.Ascii;
+
+namespace MicroscopeLaserAF
+{
+    public class Objective
+    {
+        public Objective(short initialObjectiveNumber)
+        {
+            _index = initialObjectiveNumber;
+        }
+
+        public int InFocusRange => _inFocusRange;
+
+        public float SlopeInMicrometers { get => _slopeInMicrometers; private set => _slopeInMicrometers = value; }
+
+        public int SensorRange => _sensorRange;
+
+        public void SetFocus(Axis focusAxis)
+        {
+            ATF.ATF_Make0();
+            // _focusOffset = focusAxis.GetPosition(Units.Length_Micrometres); // FIXME: This line had no effect.
+            return;
+        }
+
+        public bool SetAll(int objectiveNumber)
+        {
+            _index = (short)objectiveNumber;
+            int ecode = 0;
+            ecode += ATF.ATF_ReadMagnification(objectiveNumber, out _mag);
+            ecode += ATF.ATF_ReadInfocusRange(objectiveNumber, out _inFocusRange);
+            ecode += ATF.ATF_ReadSlopeUmPerOut(objectiveNumber, out _slopeInMicrometers);
+            ecode += ATF.ATF_ReadLinearRange(objectiveNumber, out _sensorRange);
+            return (ecode == 0); // No errors
+        }
+
+        public bool MeasureSlope(Axis focus)
+        {
+            Console.WriteLine("Move to best focus manually, hit any key to proceed");
+            Console.ReadKey();
+            ATF.ATF_Make0();
+            double pos = focus.GetPosition(Units.Length_Micrometres);
+            double delta = InFocusRange * SlopeInMicrometers * 0.2;
+            List<float> slopes = new List<float>();
+            List<double> positions = new List<double>();
+            focus.MoveAbsolute(pos - 2 * InFocusRange * SlopeInMicrometers, Units.Length_Micrometres);
+            for (int i = 1; i < 20; i++)
+            {
+                focus.MoveRelative(delta, Units.Length_Micrometres);
+                ATF.ATF_ReadPosition(out var measured);
+                var deltaPos = focus.GetPosition(Units.Length_Micrometres) - pos;
+                float newSlope = (float)(deltaPos) / measured;
+                if (Math.Abs(newSlope) > 0.09)
+                {
+                    slopes.Add(newSlope);
+                    Console.WriteLine("{0}um/DN @ {1} um", newSlope, focus.GetPosition(Units.Length_Micrometres) - pos);
+                }
+            }
+
+            float avgSlope = slopes.Average();
+            Console.WriteLine("New average: {0} Current slope: {1} um/DN", avgSlope, SlopeInMicrometers);
+            Console.WriteLine("Press y to overwrite with new measured slope");
```

Comment:
```suggestion
            Console.WriteLine("Enter y to overwrite with new measured slope");
```

Because Console.ReadLine() requires you to press enter. There is also ReadKey() which doesn't, but it also doesn't move the cursor to the next line so you'd have to do a WriteLine() to keep the next message clean.